packages/lesswrong/components/users/UsersNameDisplay.tsx

```diff
@@ -51,12 +52,30 @@ const UsersNameDisplay = ({user, nofollow=false, simple=false, classes, tooltipP
 }) => {
   const {eventHandlers} = useHover({pageElementContext: "linkPreview",  pageSubElementContext: "userNameDisplay", userId: user?._id})
 
+  const {results} = useMulti({
```

Comment:
Not sure whether the performance of this query is okay. `UsersNameDisplay` has a *lot* of copies on the front page and on post pages, and doesn't normally involve a query like this.

Reply:
The obvious speedup would be to move the query into a resolver, `goodheartRank`, and have the resolver use loaders to ensure the top-15 is only queried once.

Reply:
Yeah, that was one of the things I wanted to check tonight. Alternatively we can add either a denormalized field that saves the rank, or we can have a custom resolver, or we can pass down some props, maybe with context. Just all a bit more annoying. Was planning to do some profiling in the next half hour.